### PR TITLE
Show correct number of registrations

### PR DESCRIPTION
**Pull request by mic4ael**

### CODE REVIEW

File: indico/modules/events/registration/models/forms.py [hunk 1]
```diff
@@ -364,6 +364,12 @@ def sender_address(self):
         contact_email = self.event.contact_emails[0] if self.event.contact_emails else None
         return self.notification_sender_address or contact_email
 
+    @property
```
Comment (ThiefMaster) [inline indico/modules/events/registration/models/forms.py]: I don't like the idea of a property triggering a query on every access.

I'd either make this a `column_property` (can be deferred to not run it all the time) or a normal method (`get_registration_count()`)

Comment (ThiefMaster) [reply indico/modules/events/registration/models/forms.py]: I just saw that the other property is already done as a column property, so just do the same for the new one: https://github.com/indico/indico/blob/211375ede936dd5f2ad9dc7b2e4d15ad1b94a050/indico/modules/events/registration/models/forms.py#L418-L423

**Comment by mic4ael** [comment]
Updated ;)

File: CHANGES.rst [hunk 2]
```diff
@@ -47,6 +47,7 @@ Bugfixes
 - Fix paper reviewers getting notifications even if their type of reviewing
   has been disabled (:issue:`3852`)
 - Correctly handle merging users in the paper reviewing module (:issue:`3895`)
+- Show correct number of registrations (:issue:`#3935`)
```
Comment (ThiefMaster) [inline CHANGES.rst]: without the `#`

Comment (ThiefMaster) [reply CHANGES.rst]: "....in management area" ?

File: indico/modules/events/registration/models/forms.py [hunk 3]
```diff
@@ -415,3 +415,7 @@ def _mappers_configured():
              .where((Registration.registration_form_id == RegistrationForm.id) & Registration.is_active)
              .correlate_except(Registration))
     RegistrationForm.active_registration_count = column_property(query, deferred=True)
+    query = (select([db.func.count(Registration.id)])
```
Comment (ThiefMaster) [inline indico/modules/events/registration/models/forms.py]: empty line before?

File: CHANGES.rst (hunk 2 above)
Comment (mic4ael) [reply CHANGES.rst]: It is not only management area. There is one place in the display area where we show the number of registrations.

File: indico/modules/events/registration/templates/display/regform_list.html [hunk 4]
```diff
@@ -44,7 +44,7 @@
                     {% if any_published_registrations %}
                         {% if regform.publish_registration_count %}
                             <td class="icon-user">
-                                {{- regform.active_registration_count -}}
+                                {{- regform.existing_registrations_count -}}
```
Comment (ThiefMaster) [inline indico/modules/events/registration/templates/display/regform_list.html]: I'm not sure if this one should be changed. Why would anyone but a manager care about a rejected registration? Such a person is not participating in the event and also not counting against the limit for example!

Comment (mic4ael) [reply indico/modules/events/registration/templates/display/regform_list.html]: Yeah, I wasn't really sure about this one. Updated.

**Pull request closed without merging**